src/main/java/uk/gov/ons/ctp/response/action/service/impl/CaseNotificationServiceImpl.java

```diff
@@ -47,16 +47,23 @@ public void acceptNotification(final CaseNotification notification) throws CTPEx
     final ActionPlan actionPlan = actionPlanRepo.findById(actionPlanId);
     final UUID caseId = UUID.fromString(notification.getCaseId());
     final UUID collectionExerciseId = UUID.fromString(notification.getExerciseId());
-    final UUID partyId = UUID.fromString(notification.getPartyId());
+    UUID partyId = null;
+    if (notification.getPartyId() != null) {
+      partyId = UUID.fromString(notification.getPartyId());
+    }
+    final UUID sampleUnitId = UUID.fromString(notification.getSampleUnitId());
```

Comment:
Will this fail for those actions before the sampleunitid was added? Could you write a test to check?

Reply:
Done and done